### PR TITLE
fix(requirements): Remove tight django-treebeard restriction

### PR DESCRIPTION
**Pull request by DmytroLitvinov**
## Description

Resolves https://github.com/django-cms/djangocms-googlemap/issues/102

## Related resources

* https://github.com/django-cms/djangocms-snippet/pull/94
* https://github.com/django-cms/django-cms/issues/7476#issuecomment-1708857358

## Checklist

* [X] I have opened this pull request against ``master``
* [] I have added or modified the tests when changing logic
* [X] I have followed [the conventional commits guidelines](https://www.conventionalcommits.org/) to add meaningful information into the changelog
* [X] I have read the [contribution guidelines ](https://github.com/django-cms/django-cms/blob/develop/CONTRIBUTING.rst) and I have joined #workgroup-pr-review on 
[Slack](https://www.django-cms.org/slack) to find a “pr review buddy” who is going to review my pull request.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 REQUIREMENTS = [
     'django-cms>=3.7',
     'django-filer>=1.7',
-    'django-treebeard>=4.3,<4.5',
+    'django-treebeard>=4.3',
```
Comment (fsbraun) [inline setup.py:10]:
```suggestion
```
Let's remove this line altogether. djangocms-googlemaps does not directly depend on django-treebeard.

**Comment by fsbraun** [comment]
@DmytroLitvinov Also, for the tests to pass, you might want to add tests for django 3.2 and 4.2, django CMS 3.9, 3.10, 3.11. I know that's a bit tedious, but would help the package immensely.

File: setup.py (hunk 1 above)
Comment (DmytroLitvinov) [reply setup.py:10]: @fsbraun , I understand that djangocms-googlemap can't be used without djangocms, but still we have a reference in that library to `django-treebeard` here - https://github.com/django-cms/djangocms-googlemap/blob/42a5b81dcfd69a3559434e23772c6c8fcda0cae9/djangocms_googlemap/migrations/0005_create_nested_plugins.py#L7C1-L7C1.

So should I remove it or leave it?

**Comment by DmytroLitvinov** [comment]
@fsbraun , I will remove Python 3.7 from testing cause it is end-of-life for this version (no more even security releases) so we are not interested in releasing new version supporting it.

I think we should release a new version of package with Django like 3.2, 4.2 support and DjangoCMS 3.11, 4 (like all other libraries in ecosystem of Django)

**Comment by fsbraun** [comment]
You'll have to pin django-treebeard to <4.5 for django CMS 3.7 for it to pass the tests.

**Comment by fsbraun** [comment]
@DmytroLitvinov django CMS 3.8 needs not run on django 3.2. You can pass the test for django CMS 3.9 if you adapt the test for migrations to only look for `djangocms_googlemap` migrations.

**Comment by DmytroLitvinov** [comment]
@fsbraun , maybe lets' just support the latest version as I proposed earlier?)

It is more adding backport in requirements :)

**Review by fsbraun — APPROVED** [review]

**Comment by fsbraun** [comment]
One more thing: Is there any cost to adding "3.7", "3.10", and "3.11" to the python versions. We'll discontinue them once we introduce django CMS v4 compatibility with squashed migrations?

**Comment by DmytroLitvinov** [comment]
@fsbraun , I suppose it will run with all version of `requirements` folder packages? Like Python3.11 with Django 2.2 and DjangoCMS 3.7.

**Comment by DmytroLitvinov** [comment]
@fsbraun , pushed new changes

**Pull request merged**